app/src/org/commcare/activities/connect/ConnectActivity.java

```diff
@@ -2,17 +2,22 @@
 
 import android.os.Bundle;
 
+import androidx.fragment.app.Fragment;
+import androidx.navigation.fragment.NavHostFragment;
+
 import org.commcare.activities.CommCareActivity;
 import org.commcare.dalvik.R;
+import org.commcare.fragments.connect.ConnectDownloadingFragment;
+import org.commcare.tasks.ResourceEngineListener;
 
 import javax.annotation.Nullable;
 
-public class ConnectActivity extends CommCareActivity<ConnectActivity> {
+public class ConnectActivity extends CommCareActivity<ResourceEngineListener> {
```

Comment:
I noticed this is the only time we break the pattern that looks like:
ExampleActivity extends CommCareActivity<ExampleActivity>

In this case the generic passed to the CommCareActivity doesn't match the extending activity. Maybe that isn't an issue, but it seems like a possible source of confusion in the future if this is the only case that differs from the pattern. Your thoughts?

Reply:
Good point, I think it's a consequence of using the Nav component where we can only have one activity with more functional code in the Fragment. If we choose  to implement the `ResourceEngineListener` in  `ConnectActivity` instead we will need to define all corresponding install result handling in the ConnectActivity with a bunch of calls like `findFragment().updateStatus()` which I was trying to avoid. I believe it's a good thing for `ResourceEngineListener` interface to be decoupled with `CommCareActivity` framework in general but definitely might need more changes in future if `ConnectActivity` hosts multiple async tasks necessitating the activity itself as a receiver.